sql/updates/0.4.2--0.5.0.sql

```diff
@@ -246,3 +246,42 @@ DROP FUNCTION IF EXISTS _timescaledb_internal.drop_trigger_on_all_chunks(INTEGER
 DROP FUNCTION IF EXISTS _timescaledb_internal.get_general_trigger_definition(regclass);
 DROP FUNCTION IF EXISTS _timescaledb_internal.get_trigger_definition_for_table(INTEGER, text);
 DROP FUNCTION IF EXISTS _timescaledb_internal.need_chunk_trigger(int, oid);
+
```

Comment:
Can you explain the update strategy a bit more? I'm a little confused why you have only changed the 0.4.2-->0.5.0 update script. Say I'm running version 0.8.0 (some arbitrary version in between 0.5.0 and 0.11.0), then these aggregates are already (incorrectly) installed. How am I going to get your fix?

Reply:
Sorry, should have been clearer about that. Maybe I should add this as comment in the commit.

Aggregates are not ever incorrectly installed, at least that is not what this fixes.
Previously, if you had 0.8 installed and had some dependancies on `first`, `last` and `histogram`,  then update would fail because update script would try to do 
DROP AGGREGATE ... (see current master branch of bookend.sql)
But there is no reason to drop and recreate aggregates every time as to modify their behaviour.Wwe can CREATE AND REPLACE the functions these aggregates refer too.
So if you were to update from 0.8.0 aggregates would not be recreated as they are created once in aggregates.sql when you _install_ timescale for the first time but histogram.sql and bookend.sql would recreate all the functions the aggregate refers. But there is no problem here because histogram.sql and bookend.sql would do CREATE OR REPLACE which does not throw an error in case of dependancies.

I changed that specific update file because that is when histogram was introduced. So if you were to install a version prior to 0.5.0, you would not CREATE AGGREGATE histogram.
but because aggregate.sql is run once in install, changing it would not suffice for an update.
That is why I create the aggregate in the update file. 

It is a little ugly, but I also have to copy-paste code to create all the dependency functions. This is because update scripts are run before 0.5.0 changed sql scripts and so if I were to just do CREATE AGGREGATE, the state and transition functions inside the aggregate would not have been created and would not be recognized.

I have not modified any update scripts for `first`, `last` because they were introduced in 0.0.4 or so and the first update script is at 0.1.0.